examples/migrationExample/src/main/java/io/realm/examples/realmmigrationexample/model/Migration.java

```diff
@@ -45,32 +50,32 @@ public long execute(Realm realm, long version) {
             // Version 1
             class Person
                 @Required
-                String fullName;        // combine firstName and lastName into single field.
+                String fullName;            // combine firstName and lastName into single field.
                 int age;
-        */
-
+        ************************************************/
         // Migrate from version 0 to version 1
-        if (version == 0) {
-            Table personTable = realm.getTable(Person.class);
-
-            long fistNameIndex = getIndexForProperty(personTable, "firstName");
-            long lastNameIndex = getIndexForProperty(personTable, "lastName");
-            long fullNameIndex = personTable.addColumn(RealmFieldType.STRING, "fullName");
-            for (int i = 0; i < personTable.size(); i++) {
-                personTable.setString(fullNameIndex, i, personTable.getString(fistNameIndex, i) + " " +
-                        personTable.getString(lastNameIndex, i));
-            }
-            personTable.removeColumn(getIndexForProperty(personTable, "firstName"));
-            personTable.removeColumn(getIndexForProperty(personTable, "lastName"));
-            version++;
+        if (oldVersion == 0) {
+            RealmObjectSchema personSchema = schema.get("Person");
+
+            // Combine 'firstName' and 'lastName' in a new field called 'fullName'
+            personSchema
+                    .addField("fullName", String.class, FieldAttribute.REQUIRED)
+                    .forEach(new RealmObjectSchema.Transformer() {
```

Comment:
I guess `transform` is more explicit and RealmObjectSchema.Function has precedence in both Guava and RxJava. I'm don't have any strong preference for the current naming, so I'm fine changing it to:

```
.transform(new RealmObjectSchema.Function() {
  @Override
  public void apply(DynamicRealmObject obj) {
    // ...
  }
})
```

?